localstack-core/localstack/services/lambda_/event_source_mapping/pollers/stream_poller.py

```diff
@@ -87,6 +91,11 @@ def failure_payload_details_field_name(self) -> str:
     def get_approximate_arrival_time(self, record: dict) -> float:
         pass
 
+    @abstractmethod
+    def format_datetime(self, time: datetime) -> str:
+        """Formats a datetime in the correct format for DynamoDB (with ms) or Kinesis (without ms)"""
```

Comment:
nice catch 🕵️ and clearer implementation as well as documentation (`[:-3]` 🧹 )

I guess this one fixes:
```
"$..Messages..Body.KinesisBatchInfo.approximateArrivalOfFirstRecord"
"$..Messages..Body.KinesisBatchInfo.approximateArrivalOfLastRecord"
```